.github/workflows/format.yml

```diff
@@ -1,17 +1,27 @@
-name: Check format
+name: format
 
 on:
   pull_request:
     branches:
       - master
-  push:
-    branches:
-      - master
 
 jobs:
-  check-format:
+  format:
     runs-on: ubuntu-latest
-    name: Check format
     steps:
-      - uses: actions/checkout@v2
-      - uses: ory/prettier-styles@v1.0.1
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Cache node modules
```

Comment:
Could we simplify this with https://github.com/actions/setup-node#caching-global-packages-data ?

Reply:
Not really, and I was surprised by how useless the `cache` option in the `setup-node` action is! That option only caches the downloaded `.tar.gz` files [somewhere else](https://docs.npmjs.com/cli/v6/using-npm/config#cache). Even with this cache in place, npm still has to create `node_modules` on each CI run. Which somehow takes up to 2 minutes for more complex repos like ory/web. Even if it doesn't need to download the modules again. Even if the entire `node_modules` folder already exists and it wouldn't need to do anything!

You can verify this by running `npm i` in ory/web twice on your machine. The second run (in which it has a fully populated global cache) still takes a very long time.

So, instead I decided to cache the `node_modules` folder directly, using the `cache` action. If CI finds a cached version matching the exact content of `package-lock.json`, it doesn't even run `npm install` at all because it has everything it needs to run, i.e. a fully set up `node_modules` folder. If `package-lock.json` is changed, it does a normal `npm install`.

I guess the way the `setup-node` action caches things is optimized for being super generic totally bulletproof, at the cost of not being able to do more opinionated caching. Which is probably the right thing for such a generic action.